### DISCUSSION TITLE
Should we include images from `Readme` in `.vsix` file?

### DISCUSSION THREAD

**Discussion by oliversen** [microsoft/vscode-discussions#1191] [Extension Development QnA] [answered]
Should we include images from `Readme` in `.vsix` file?

**Comment by alefragnani** [comment, marked as answer]
Hi @oliversen ,

There is no _guideline/recomendation_ describing if you should or should not use include it but. The only warning I would point you is about SVG images, which `vsce` will check while packaging/publishing. You can find more details in [Publishing extensions](https://code.visualstudio.com/api/working-with-extensions/publishing-extension#publishing-extensions). Personally, I use both, embedded and external images, depending on their objectives. And have no issues.

I'm not sure the _size_ of the `vsix` file is something you are worried about, or the need for offline distribution, but these are variables that you may also consider while deciding whether to include the images or not.

Hope this helps

**Comment by abrilurena** [comment]
Hi, @oliversen! do you still need help with this?

**Reply by ssbarnea** [reply]
The problem is quite valid and it needs some clarification.

People leaned that can rely on automatic mapping of relative urls to github but they forgot that this points to head of repository. This means that removing or renaming images on the code will break rendering of already released extensions.

Even updating an image/video can cause problems as a new feature might be depicted which was not present in the current release of the extension.

Also there is no clear documentation regarding which formats are supported or not, webp, webm, mp4...
